core/src/main/java/io/kestra/core/models/tasks/NamespaceFilesInterface.java

```diff
@@ -0,0 +1,14 @@
+package io.kestra.core.models.tasks;
+
+import io.kestra.core.models.annotations.PluginProperty;
+import io.swagger.v3.oas.annotations.media.Schema;
+
+public interface NamespaceFilesInterface {
+    @Schema(
+        title = "Inject namespace files",
+        description = "Inject namespace file on this task. If true, inject all namespaces files or use filter to limit injected files.",
+        anyOf = {Boolean.class, NamespaceFiles.class}
+    )
+    @PluginProperty
+    Object getNamespaceFiles();
```

Comment:
Wouldn't it be easier to use the `NamespaceFiles` object here and have a way to inject everything from it?
Like a `all` boolean or just document that to inject everything one should configure `include: **/*`?

I wonder if those multi-types that we often use are easy to use for not too technical people.

Reply:
We discussed it initially via Slack and the initial proposal was to have 2 properties: one boolean `namespaceFiles` whether to fetch all namespace files and then if true, optional `namespaceFilesFilter` property with include/exclude. Given that there is some risk of confusion of using the filter without setting the boolean to true (it must be set to false by default), we went for that option with 1 property

good point though and well spotted

Reply:
for now, a good tradeoffs might be if we add maybe 2-3 examples in the task docs to show the usage